### PR TITLE
GEOMESA-2630,GEOMESA-2631 Fix JSON output for Spark dataframes

### PR DESCRIPTION
**Pull request by elahrvivaz**
* New conversion class in spark-sql module
* Deprecated spark-jts conversion class
* Separated out classes in GeoMesaSparkSQL file

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
stacks on top of #2272

File: docs/user/spark/core.rst [hunk 1]
```diff
@@ -139,3 +139,20 @@ To save features, use the ``save()`` method:
 Note that some providers may be read-only.
 
 See :doc:`./providers` for details on specific provider implementations.
+
+
+GeoJSON Output
```
Comment (jnh5y) [inline docs/user/spark/core.rst]: Does 'core' depend on JTS?  Or is it the other way?

I do like having GeoJSON support in the most basic module.  If the docs are indicating a move of functionality from the Spark+JTS module, I'd like to consider alternatives for a minute.

Comment (elahrvivaz) [reply docs/user/spark/core.rst]: I think core depends on JTS. The old method is still there, but we don't have geotools on the classpath in the JTS module so we'd have to re-implement geojson encoding to keep it there...

Comment (elahrvivaz) [reply docs/user/spark/core.rst]: possibly we could use some combination of a generic json writer and the geojson geometry writer from jts... but I dunno seems like people would be using the core module anyway.

Comment (jnh5y) [reply docs/user/spark/core.rst]: Yeah, I wonder how hard it'd be to do that? 

I'd hate for us to pull in GeoTools for something that'd be really, really quick.

Comment (jnh5y) [reply docs/user/spark/core.rst]: Am I asking for a different feature/capability?  Aka, should this idea be a separate ticket?

Comment (elahrvivaz) [reply docs/user/spark/core.rst]: Is this an actual use case though? Who is going to be using just the JTS module and nothing else? All our spark-runtime jars bundle core up too. I'd opine it's not worth trying to implement something just in JTS, when we can easily use geotools in the other module.

Comment (jnh5y) [reply docs/user/spark/core.rst]: RasterFrames and other projects are pointed at GeoMesa's JTS Spark module.

Comment (jnh5y) [reply docs/user/spark/core.rst]: That said, it is a different issue, so I'm shipping this PR!

**Pull request merged**